File: cpp/041-050/Jump Game II.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        
        if (nums.size() < 2) return 0;
        
        int count(0);
        int step(0);
        int nextstep(0);
        
        for (int i = 0; i < nums.size(); i++) {
            if (i + nums[i] > nextstep) nextstep = i + nums[i];
            if (i == step && i != nums.size()-1) {
                step = nextstep;
                count++;
            }
        }
        
        return count;
    }
    
};

File: python/041-050/Jump Game II.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def jump(self, nums):
        step = 0
        nextstep = 0
        count = 0
        
        for i in range(len(nums)):
            if i+nums[i] > nextstep:
                nextstep = i + nums[i]
            if i == step and i != len(nums)-1:
                count += 1
                step = nextstep
        
        return count